### ISSUE TITLE
[QUESTION] How would you get artists and albums with this client, if possible?

### ISSUE THREAD

**Issue by tacohitbox** [LuanRT/YouTube.js#78]
## Question

I looked around the documentation, and I see no mention of getting albums from YT Music at the moment. Is it possible to at the moment, and if not, can you add it? 

Also you can get some artist metadata but I believe it's different than what YouTube Music responds with. Is it possible to get more information, like top tracks?

Many thanks.

## Sample Code

I tried reading the album IDs as playlists with no luck. Example below.

```js
const Yt = require("youtubei.js");

(async function() {
  let yt = await new Yt({gl: "US"});
  let a = await yt.getPlaylist("MPREb_iYZbnszRqas", {client: "YTMUSIC"});
  console.log(a);
})();
```

I did something similar with artist IDs and got some but not all of the information I need.

```js
const Yt = require("youtubei.js");

(async function() {
  let yt = await new Yt({gl: "US"});
  let a = await yt.getChannel("UCXWUt1Cu2hVZYTaJ44ddusA", {client: "YTMUSIC"});
  console.log(a);
})();
```

**Comment by tacohitbox** [comment]
Edited after I noticed I had similar trouble with artist IDs.

**Comment by LuanRT** [comment]
Not possible in v1, but v2 will bring full support for YouTube Music. See #65.

**Comment by LuanRT** [comment]
Not possible in v1, but v2 will bring full support for YouTube Music. See #65.

**Comment by LuanRT** [comment]
This is now possible after 7fc9b526b00abdbe40e4915696677111180e4ea4 & 97d4cc1056ff7305d26ac768b280a0ac67b95646. If you would like to test it simply install the dev version.

#### Examples:
```js
const album = await youtube.music.getAlbum('MPREb_lM0ql53ZzAr');
const artist = await youtube.music.getArtist('UC52ZqHVQz5OoGhvbWiRal6g');
```

**Comment by tacohitbox** [comment]
Other than a lot of errors being but into the console, like so, it works great:

```
InnertubeError: Something went wrong at LikeButton!
This is a bug, please report it at https://github.com/LuanRT/YouTube.js/issues
    at Parser.formatError (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:188:9)
    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:155:18)
    at new Menu (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/Menu.js:10:37)
    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:168:26)
    at new MusicResponsiveListItem (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/MusicResponsiveListItem.js:52:24)
    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:151:28)
    at new MusicShelf (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/MusicShelf.js:13:28)
    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:151:28)
    at new SectionList (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/SectionList.js:12:28)
    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:168:26) {
  info: {
    stack: "TypeError: Cannot read properties of undefined (reading 'map')\n" +
      '    at new LikeButton (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/LikeButton.js:16:44)\n' +
      '    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:151:28)\n' +
      '    at new Menu (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/Menu.js:10:37)\n' +
      '    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:168:26)\n' +
      '    at new MusicResponsiveListItem (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/MusicResponsiveListItem.js:52:24)\n' +
      '    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:151:28)\n' +
      '    at new MusicShelf (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/MusicShelf.js:13:28)\n' +
      '    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:151:28)\n' +
      '    at new SectionList (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/SectionList.js:12:28)\n' +
      '    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:168:26)'
  },
  date: 2022-06-21T05:47:11.036Z,
  version: '1.4.3'
}
```

**Comment by tacohitbox** [comment]
The artist suggested, `UC52ZqHVQz5OoGhvbWiRal6g`, and all others I tried, gives this error: 
```
TypeError: Cannot read properties of undefined (reading 'url_canonical')
    at new Album (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/ytmusic/Album.js:22:39)
    at Music.getAlbum (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/core/Music.js:88:12)
```

**Comment by tacohitbox** [comment]
Many thanks, by the way, for helping me out :)

**Comment by LuanRT** [comment]
> The artist suggested, `UC52ZqHVQz5OoGhvbWiRal6g`, and all others I tried, gives this error:
> 
> ```
> TypeError: Cannot read properties of undefined (reading 'url_canonical')
>     at new Album (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/ytmusic/Album.js:22:39)
>     at Music.getAlbum (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/core/Music.js:88:12)
> ```

Seems like you've used `getAlbum()` when it should be `getArtist()`.

**Comment by LuanRT** [comment]
> Other than a lot of errors on albums being but into the console, like so, it works great:

Thanks for reporting that.
Those are silent errors so you don't have to worry about them for now.

**Comment by tacohitbox** [comment]
> > The artist suggested, `UC52ZqHVQz5OoGhvbWiRal6g`, and all others I tried, gives this error:
> > ```
> > TypeError: Cannot read properties of undefined (reading 'url_canonical')
> >     at new Album (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/ytmusic/Album.js:22:39)
> >     at Music.getAlbum (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/core/Music.js:88:12)
> > ```
> 
> Seems like you've used `getAlbum()` when it should've been `getArtist()`.

Silly mistake on my part, apologies!

**Comment by tacohitbox** [comment]
```
InnertubeError: Something went wrong at MusicResponsiveListItem!
This is a bug, please report it at https://github.com/LuanRT/YouTube.js/issues
    at Parser.formatError (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:188:9)
    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:155:18)
    at new MusicShelf (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/MusicShelf.js:13:28)
    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:151:28)
    at new SectionList (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/SectionList.js:12:28)
    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:168:26)
    at new Tab (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/Tab.js:13:27)
    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:151:28)
    at get tabs [as tabs] (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/TabbedSearchResults.js:15:19)
    at new Search (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/ytmusic/Search.js:27:37) {
  info: {
    stack: "TypeError: Cannot read properties of undefined (reading 'page_type')\n" +
      '    at new MusicResponsiveListItem (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/MusicResponsiveListItem.js:28:34)\n' +
      '    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:151:28)\n' +
      '    at new MusicShelf (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/MusicShelf.js:13:28)\n' +
      '    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:151:28)\n' +
      '    at new SectionList (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/SectionList.js:12:28)\n' +
      '    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:168:26)\n' +
      '    at new Tab (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/Tab.js:13:27)\n' +
      '    at Parser.parse (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/index.js:151:28)\n' +
      '    at get tabs [as tabs] (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/contents/classes/TabbedSearchResults.js:15:19)\n' +
      '    at new Search (/home/a/Documents/git/mollusk/node_modules/youtubei.js/lib/parser/ytmusic/Search.js:27:37)'
  },
  date: 2022-06-22T03:45:46.299Z,
  version: '1.4.3'
}
```

got another error for ya, while searching `no copyright sounds` on youtube music @LuanRT

**Comment by LuanRT** [comment]
> got another error for ya, while searching `no copyright sounds` on youtube music @LuanRT

Thanks! Should be fixed now (20556970a70ef28fbb473aec6452818fc4c1bc07).
